### PR TITLE
feat: improve callback value for webContents.print()

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Resolves https://github.com/electron/electron/issues/18980.

Adds granularity to the optional callback in `webContents.print()` by adding a `failureType` value in addition to the `success` boolean that differentiates between `cancelled` and `failed` print jobs.

Tested with [this repo](https://github.com/popod/electron-quick-start/tree/print-bug-1).

cc @deepak1556 @brenca 

also cc @popod as i know you're tracking printing work :)

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Improved `webContents.print()` callback value to distinguish between success, failure, and cancellation.

### CODE REVIEW

**Comment by popod** [comment]
I think this is a "little" breaking change and that it should be indicated in the release notes.

- with older electron versions `webContents.print()` return `false` in case of errors
- with newer electron versions `webContents.print()` return `failed` which could be interpreted as `true` (but we attempt to have `false` with older electron logic)

**Comment by codebytere** [comment]
@popod yeah, this wouldn't be backported and the upcoming change would be indicated in notes and code as ability permits!

**Comment by popod** [comment]
Juste for informations, @MarshallOfSound proposal would solve #15568

**Comment by MarshallOfSound** [comment]
> And why not backporting this to v6 which is still in beta for now

@popod We follow semver and this being a feature would be `semver/minor`, they can't be backported to stable release lines.  Admitting a feature to an existing beta release line requires a discussion + good reasoning + a vote in the Releases WG.  With 6 so close to the end of its beta cycle it's unlikely any more features will be admitted to destabilization.

File: patches/chromium/printing.patch [hunk 1]
```diff
@@ -204,19 +214,27 @@ index 7ba43aada1ac44827cca264d6f37814e4a91f458..c41b0c24974147e847baa21b9b489261
...
++#include "v8/include/v8.h"
+ 
+ struct PrintHostMsg_DidPrintDocument_Params;
+ 
+@@ -39,6 +40,8 @@ class PrintJob;
  class PrintQueriesQueue;
  class PrinterQuery;
  
-+using CompletionCallback = base::OnceCallback<void(bool)>;
++using CompletionCallback = base::OnceCallback<void(bool, v8::Local<v8::Value>)>;
```
Comment (zcbenz) [inline patches/chromium/printing.patch]: Should just pass `const std::string&`, V8 conversation would be done automatically.

**Pull request merged**